EventsExpress/ClientApp/src/actions/register/register-complete-action.js

```diff
@@ -1,32 +1,47 @@
+import { SubmissionError } from 'redux-form';
+import moment from 'moment';
+import jwt from 'jsonwebtoken';
 import { AuthenticationService } from '../../services';
-import { createBrowserHistory } from 'history';
-import { setSuccessAllert } from '../alert-action';
+import { setSuccessAllert, setErrorAllertFromResponse } from '../alert-action';
 import { buildValidationState } from '../../components/helpers/action-helpers';
-import { SubmissionError } from 'redux-form';
 import { jwtStorageKey } from '../../constants/constants';
-import jwt from 'jsonwebtoken';
+import { getUserInfo } from '../login/login-action';
 
 const api_serv = new AuthenticationService();
-const history = createBrowserHistory({ forceRefresh: true });
 
-export default function registerComplete(data) {
+export default function registerComplete(data, { shouldSaveMoreInfo }) {
     return async dispatch => {
-        data.accountId = getAccountIdFromJWT()
+        const body = {
+            ...data,
+            birthday: moment.utc(data.birthday).local().format('YYYY-MM-DD[T00:00:00]'),
+            accountId: getAccountIdFromJWT(),
+        };
 
-        let response = await api_serv.setRegisterComplete(data);
-        if (!response.ok) {
-            throw new SubmissionError(await buildValidationState(response));
+        const profileResponse = await api_serv.setRegisterComplete(body);
+        if (!profileResponse.ok) {
+            dispatch(setErrorAllertFromResponse(profileResponse.clone()));
+            throw new SubmissionError(await buildValidationState(profileResponse));
         }
-        let jsonRes = await response.json();
+        
+        const jsonRes = await profileResponse.json();
         localStorage.setItem(jwtStorageKey, jsonRes.token);
+
+        if (shouldSaveMoreInfo === true) {
+            const moreInfoResponse = await api_serv.setMoreInfo(data);
+            if (!moreInfoResponse.ok) {
+                dispatch(setErrorAllertFromResponse(moreInfoResponse.clone()));
+                throw new SubmissionError(await buildValidationState(moreInfoResponse));
+            }
+        }
+
+        dispatch(getUserInfo());
         dispatch(setSuccessAllert('Your profile was updated'));
-        dispatch(history.push('/home'))
```

Comment:
Why don't we navigate to home afterwards?

Reply:
This was the original logic where the user was navigated to home right after submitting the form successfully. Now there is a success page with links to profile and events pages